usaspending_api/database_scripts/matview_generator/tas_autocomplete_matview.json

```diff
@@ -0,0 +1,58 @@
+{
+  "final_name": "tas_autocomplete_matview",
+  "refresh": true,
+  "matview_sql": [
+    "select",
+    "    min(treasury_account_identifier) tas_autocomplete_id,",
```

Comment:
min? Why min?

Reply:
I needed `allocation_transfer_agency_id`, `agency_id`, `beginning_period_of_availability`, `ending_period_of_availability`, `availability_type_code`, `main_account_code`, and `sub_account_code` to be unique so I group by those.  Django really wants a primary key on the model so to cheese that I just took the minimum id for the TAS.  Let me know if this doesn't make sense and I'll esplain more in person.

Reply:
So the idea is that no two combinations of the other elements will ever have the same TAS (although not necessarily the other way), so it's safe to just grab any old TAS id to the PK? Wouldn't this introduce risk if duplicates were added to the TAS table for whatever reason? We've had to do that for agencies before...

Reply:
We talked about this a little yesterday, but just to close the loop, this query guarantees that each `tas_autocomplete_matview` row will have a unique `tas_autocomplete_id` and that that id will be unique for each distinct grouping of TAS components.  The `min` only applies to the grouping so it will be the `min` `treasury_account_identifier` for the grouping of TAS components as found in the `treasury_appropriation_account` table.